lib/AbstractHandle/Utils/Handler.py

```diff
@@ -39,7 +39,7 @@ def _process_handle(self, handle, user_id):
 
         handle = {k: v for k, v in handle.items() if k in self.FIELD_NAMES}  # remove unnecessary fields
         if not handle.get('hid'):
-            hid_counter = self.mongo_util.get_hid_counter()
+            hid_counter = self.mongo_util.increase_counter()
             handle['hid'] = int(hid_counter)
         else:
```

Comment:
Actually - does this part of the code mean that the user can specify the handle ID? Because that seems very bad if they specify an ID that hasn't been assigned yet.

Reply:
yea. In the old HS, it's allowed. Should we raise or just ignore user-specified handle ID? It's mostly for update handle with persist_handle. Since we now remove that logic, I can remove this as well.

Reply:
I'd prefer raise myself - is there any reason not to do that?

Reply:
nope. I'd rather raise too.